.clang-tidy

```diff
@@ -0,0 +1,16 @@
+---
+Checks: 'clang-diagnostic-*,
```

Comment:
Note also that i tried to construct an analysis list that balanced "too annoying" with "not helpful enough".

It's possible to disable some of the more frequent analyzer reports by making further modifications to this file.